packages/client/src/components/ReviewDialog/ReviewDialog.jsx

```diff
@@ -10,30 +10,56 @@ import { Avatar } from '@mui/material';
 import { RatingStars } from '../RatingStars/RatingStars';
 import styled from '@emotion/styled';
 import { apiURL } from '../../apiURL';
-import fallBackMovies from '../../assets/fallBackMovies.json';
 
-export function ReviewDialog({ initialState, handleClose, user, movieId }) {
-  const [formData, setFormData] = useState(fallBackMovies);
+export function ReviewDialog({
+  initialState,
+  handleClose,
+  currentReview,
+  movieId,
+  currentUserId,
+}) {
+  const [review, setReview] = useState(currentReview);
+  const [formData, setFormData] = useState(currentReview);
   useEffect(() => {
-    (async () => {
-      if (movieId) {
-        try {
-          const response = await fetch(
-            // Here I use A fixed uid value 123543d.This account doesn't have any reviews.When we realize the function of adding new reviews,I can use user.uid instaed
-            `${apiURL()}/reviews/${movieId}/uid/123543d`,
-          );
-          if (response.ok) {
-            const data = await response.json();
-            if (data) {
-              setFormData(data);
-            }
-          }
-        } catch (error) {
-          throw new Error(error);
-        }
-      }
-    })();
-  }, [movieId]);
+    setFormData(currentReview);
+    setReview(currentReview);
+  }, [currentReview]);
+  const postValue = {
+    movie_id: movieId,
+    user_id: currentUserId,
+    rating: formData ? formData.rating : '',
+    review_text: formData ? formData.review_text : '',
+  };
+
+  const deleteReview = async (id) => {
+    await fetch(`${apiURL()}/reviews/${id}`, {
+      method: 'DELETE',
+    });
+  };
+
+  const updateReview = async (id) => {
+    const body = formData;
+    await fetch(`${apiURL()}/reviews/${id}`, {
+      method: 'PATCH',
+      body: JSON.stringify(body),
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+    });
+  };
+
+  const postReview = async () => {
+    const body = postValue;
+    await fetch(`${apiURL()}/reviews/`, {
+      method: 'POST',
+      body: JSON.stringify(body),
+      headers: {
```

Comment:
In this case no destructuring needed, because new object was created for postValue